### PR TITLE
Refactor order of gradient computations and model updates to make compatible with later versions of Pytorch

### PR DESCRIPTION
**Pull request by Aladoro**
Fixes incompatibility with recent versions of Pytorch (tested with torch==1.8.1), where original order of gradient computations/weight updates caused a Runtime error, due to _alpha_ being updated before _policy_loss.backward()_ is called.

Otherwise, none of the behavior should be affected.

Thanks for sharing the code ^^

### CODE REVIEW

**Review by watchernyu — APPROVED** [review]
@Aladoro Thank you very much for sending the pull request! I will now merge the changes.
